src/VisualStudio/Core/Impl/CodeModel/ExternalElements/ExternalCodeProperty.cs

```diff
@@ -123,23 +116,61 @@ public EnvDTE80.vsCMOverrideKind OverrideKind
         {
             get
             {
-                throw new NotImplementedException();
+                var symbol = PropertySymbol;
+                var result = EnvDTE80.vsCMOverrideKind.vsCMOverrideKindNone;
+
+                if (symbol.IsAbstract)
+                {
+                    result |= EnvDTE80.vsCMOverrideKind.vsCMOverrideKindAbstract;
+                }
+
+                if (symbol.IsVirtual)
+                {
+                    result |= EnvDTE80.vsCMOverrideKind.vsCMOverrideKindVirtual;
+                }
+
+                if (symbol.IsOverride)
+                {
+                    result |= EnvDTE80.vsCMOverrideKind.vsCMOverrideKindOverride;
+                }
+
+                if (symbol.IsSealed)
+                {
+                    result |= EnvDTE80.vsCMOverrideKind.vsCMOverrideKindSealed;
+                }
+
+                return result;
             }
 
             set
             {
-                throw new NotImplementedException();
+                throw Exceptions.ThrowEFail();
             }
         }
 
         public EnvDTE.CodeElement Parent2
         {
-            get { throw new NotImplementedException(); }
+            get { return (EnvDTE.CodeElement)base.Parent; }
         }
 
         public EnvDTE80.vsCMPropertyKind ReadWrite
         {
-            get { throw new NotImplementedException(); }
+            get
+            {
+                var symbol = PropertySymbol;
+                if (symbol.GetMethod != null)
+                {
+                    return symbol.SetMethod != null
+                        ? EnvDTE80.vsCMPropertyKind.vsCMPropertyKindReadWrite
+                        : EnvDTE80.vsCMPropertyKind.vsCMPropertyKindReadOnly;
+                }
+                else if (symbol.SetMethod != null)
+                {
+                    return EnvDTE80.vsCMPropertyKind.vsCMPropertyKindWriteOnly;
+                }
+
+                throw Exceptions.ThrowEUnexpected();
```

Comment:
Just curious: is this needed since E_UNEXPECTED is a possibly valid return and you want to be strict about the COM error, or should we throw our internal "unexpected" thing so it's very, very clear this is unexpected?